crates/torii/graphql/src/query/mod.rs

```diff
@@ -110,10 +123,20 @@ pub fn value_mapping_from_row(
                 field_name.to_string()
             };
 
-            Ok((
-                Name::new(field_name),
-                fetch_value(row, &column_name, &type_data.type_ref().to_string())?,
-            ))
+            let field_type = type_data.type_ref().to_string();
+            let value = fetch_value(row, &column_name, &field_type)?;
+            let value = match Primitive::from_str(&field_type) {
+                Ok(primitive) => match primitive {
+                    Primitive::U128(_)
+                    | Primitive::ContractAddress(_)
+                    | Primitive::ClassHash(_)
+                    | Primitive::Felt252(_) => remove_hex_leading_zeros(value),
+                    _ => value,
```

Comment:
Do we need U256 here too?

Reply:
Ah yes; nice catch!
The function changed a little to handle this cause for this type the hexadecimal string does not start with 0x:
```rust
fn remove_hex_leading_zeros(value: Value) -> Value {
    if let Value::String(mut str_val) = value {
        if !str_val.starts_with("0x") {
            str_val = str_val.trim_start_matches("0x").to_string();
        }
        let trimmed_hex = str_val.trim_start_matches('0');
        Value::String(format!("0x{:0>1}", trimmed_hex))
    } else {
        value
    }
}
```

Reply:
Hmm the hex string should always start with 0x. If that is not the case, we should fix it elsewhere

Reply:
Oh, thought it was intended; I can just add 0x here then:
```rust
Primitive::U256(_) => {
                if value.len() < 2 {
                    Err(PrimitiveError::NotEnoughFieldElements)
                } else {
                    let mut buffer = [0u8; 32];
                    let value0_bytes = value[0].to_bytes_be();
                    let value1_bytes = value[1].to_bytes_be();
                    buffer[..16].copy_from_slice(&value0_bytes);
                    buffer[16..].copy_from_slice(&value1_bytes);
                    Ok(format!("'{}'", hex::encode(buffer)))
                }
            }
```

Reply:
Does it make sense to have this constant defined in `dojo-types` & use it in `torii` later?
```rust
pub const HEXADECIMAL_STR_PREFIX: &str = "0x";
```
Maybe it's just not needed

Reply:
Nah I think a constant in this case is a bit overkill

Reply:
Updated!